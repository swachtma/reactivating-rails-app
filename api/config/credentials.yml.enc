NruSqAfDHjCumPIyJn6xsY9LOZuE3ZzX2o/1QTdVkugVslYlIp4h/gmrnNPKDUKYZ8vGEGHPJ5EwlOSClaonGjgUB4yUG4tz8I0T++XLbiEU8qhxATlTf+s9JNfusXdgIB2ZFO4ABSDKhxhzPLbHYJVvLghx3WmwMMpO+GBgzlp94DLCX2IR38jeHEydFPy73oteodKVbTBp0g91v8DhpnZUTWHYb0JdqS+otrdrS1mar9MlcMnTpb8chvLisrrvE0SR/QUagAbd5UD/d02NczFYmAdKsyztbryVyakhgosZhCJFY3KNwxayxDC3sF5xFQHgli8qJNXNAhzaYDVf+n7D9k6g9BHTV5YRXdICgYORVWo+a9EhjBTmVwQsJN+IrdZaqxQVoa2DMEA3CVjOlp1wtLNwHbHArmId--8+ArDlMZvB3FEYTs--m4O75Mk62PJos0rR8zw91Q==