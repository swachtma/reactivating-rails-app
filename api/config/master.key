dc4eeb306d00d2a5e6993b94e449787e