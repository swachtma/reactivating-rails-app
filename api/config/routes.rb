Rails.application.routes.draw do
  scope :api do
    get 'github/', to: "authentication#github", format: false
    get 'hydrate_user/', to: "authentication#show"
    get 'nodes/', to: "nodes#index"
    get 'chapters/', to: "chapters#index"
    resources :bookmarks, only: [:index, :create]
  end
end