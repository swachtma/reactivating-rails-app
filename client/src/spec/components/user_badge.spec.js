/* global expect, jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import config from '../../test_helper';
import UserBadge, { ProfileIcon } from '../../components/user_badge';

describe("<UserBadge />", () =>{
  it("renders GithubIcon on !props.username", ()=>{
    let wrapper = shallow(<UserBadge />);
    expect(wrapper.find("Connect(GithubIcon)").exists()).toBe(true);
  });
  
  it("renders ProfileIcon on props.username", ()=>{
    let wrapper = shallow(<UserBadge username="TestDude" />);
    expect(wrapper.find("ProfileIcon").exists()).toBe(true);
  });
});

describe("<ProfileIcon />",()=>{
  let mockDispatch = jest.fn();
  let mockHandler = jest.fn();
  let wrapper = shallow(
    <ProfileIcon username="TestDude" avatar="someurl.png"
      dispatchSignalSignout={mockDispatch} handleSidebarVisibility={mockHandler} />
  );
  
  it("calls function passed in dispatchSignalSignout on Signout click",()=>{
    let link = wrapper.find("DropdownItem");
    link.simulate("click");
    expect(mockDispatch).toBeCalled();
  });
  
  it("calls function passed in handleSidebarVisibility on focus",()=>{
    let menubar = wrapper.find("Dropdown");
    menubar.simulate("focus");
    expect(mockHandler).toBeCalled();
  });
  
  it("renders according to snapshot",()=>{
    expect(wrapper).toMatchSnapshot();
  });
});