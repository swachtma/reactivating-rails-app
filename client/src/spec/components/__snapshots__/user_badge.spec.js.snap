// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileIcon /> renders according to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileIcon
    avatar="someurl.png"
    dispatchSignalSignout={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    handleSidebarVisibility={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    username="TestDude"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Dropdown
        additionLabel="Add "
        additionPosition="top"
        closeOnBlur={true}
        compact={true}
        deburr={false}
        icon={
          <Image
            as="img"
            avatar={true}
            src="someurl.png"
            style={
              Object {
                "marginLeft": "1.757px",
                "maxHeight": "32x",
                "maxWidth": "32px",
              }
            }
            ui={true}
          />
        }
        minCharacters={1}
        noResultsMessage="No results found."
        onFocus={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        openOnFocus={true}
        pointing="top right"
        renderLabel={[Function]}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        wrapSelection={true}
      >
        <DropdownMenu>
          <DropdownItem
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
          >
            Sign Out
          </DropdownItem>
        </DropdownMenu>
      </Dropdown>,
      "fitted": true,
      "name": "TestDude",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "additionLabel": "Add ",
        "additionPosition": "top",
        "children": <DropdownMenu>
          <DropdownItem
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
          >
            Sign Out
          </DropdownItem>
        </DropdownMenu>,
        "closeOnBlur": true,
        "compact": true,
        "deburr": false,
        "icon": <Image
          as="img"
          avatar={true}
          src="someurl.png"
          style={
            Object {
              "marginLeft": "1.757px",
              "maxHeight": "32x",
              "maxWidth": "32px",
            }
          }
          ui={true}
        />,
        "minCharacters": 1,
        "noResultsMessage": "No results found.",
        "onFocus": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "openOnFocus": true,
        "pointing": "top right",
        "renderLabel": [Function],
        "searchInput": "text",
        "selectOnBlur": true,
        "selectOnNavigation": true,
        "wrapSelection": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <DropdownItem
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
          >
            Sign Out
          </DropdownItem>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Sign Out",
            "onClick": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
          },
          "ref": null,
          "rendered": "Sign Out",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Dropdown
          additionLabel="Add "
          additionPosition="top"
          closeOnBlur={true}
          compact={true}
          deburr={false}
          icon={
            <Image
              as="img"
              avatar={true}
              src="someurl.png"
              style={
                Object {
                  "marginLeft": "1.757px",
                  "maxHeight": "32x",
                  "maxWidth": "32px",
                }
              }
              ui={true}
            />
          }
          minCharacters={1}
          noResultsMessage="No results found."
          onFocus={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          openOnFocus={true}
          pointing="top right"
          renderLabel={[Function]}
          searchInput="text"
          selectOnBlur={true}
          selectOnNavigation={true}
          wrapSelection={true}
        >
          <DropdownMenu>
            <DropdownItem
              onClick={
                [MockFunction] {
                  "calls": Array [
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                  ],
                }
              }
            >
              Sign Out
            </DropdownItem>
          </DropdownMenu>
        </Dropdown>,
        "fitted": true,
        "name": "TestDude",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "additionLabel": "Add ",
          "additionPosition": "top",
          "children": <DropdownMenu>
            <DropdownItem
              onClick={
                [MockFunction] {
                  "calls": Array [
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                  ],
                }
              }
            >
              Sign Out
            </DropdownItem>
          </DropdownMenu>,
          "closeOnBlur": true,
          "compact": true,
          "deburr": false,
          "icon": <Image
            as="img"
            avatar={true}
            src="someurl.png"
            style={
              Object {
                "marginLeft": "1.757px",
                "maxHeight": "32x",
                "maxWidth": "32px",
              }
            }
            ui={true}
          />,
          "minCharacters": 1,
          "noResultsMessage": "No results found.",
          "onFocus": [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "openOnFocus": true,
          "pointing": "top right",
          "renderLabel": [Function],
          "searchInput": "text",
          "selectOnBlur": true,
          "selectOnNavigation": true,
          "wrapSelection": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <DropdownItem
              onClick={
                [MockFunction] {
                  "calls": Array [
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                  ],
                }
              }
            >
              Sign Out
            </DropdownItem>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Sign Out",
              "onClick": [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              },
            },
            "ref": null,
            "rendered": "Sign Out",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
